src/fast-path.js

```diff
@@ -245,6 +245,21 @@ FPp.needsParens = function(assumeExpressionContext) {
     return true;
   }
 
+  if (
+    (node.type === "BindExpression" ||
+     node.type === "UpdateExpression" ||
+     node.type === "BinaryExpression" ||
+     node.type === "LogicalExpression" ||
+     node.type === "AssignmentExpression" ||
+     node.type === "ConditionalExpression" ||
+     node.type === "SequenceExpression") &&
+    ((parent.type === "ArrowFunctionExpression" && parent.body === node) ||
+    parent.type === "ExpressionStatement") &&
+    startsWithOpenCurlyBrace(node)
+  ) {
+    return true;
+  }
+
   switch (node.type) {
     case "CallExpression":
       if (
```

Comment:
This `if` is unnecessary; instead you should have a `|| node.type === "CallExpression"` case a few lines above. And `node.type === "TaggedTemplateExpression"`, come to think, for test case

```js
x => ({}()``)
```